### ISSUE TITLE
Allow normalization method override at inference time

### ISSUE THREAD

**Issue by rgthelen** [llama-farm/llamafarm#646]
## Summary

Currently, the normalization method (standardization, zscore, raw) is set at model training time and cannot be changed at inference time. This limits flexibility for users who want to experiment with different score interpretations without retraining.

## Current Behavior

- Normalization method is set during `fit()` and baked into the model
- The model already computes and stores ALL normalization statistics during training:
  - `_norm_median` / `_norm_iqr` for standardization
  - `_norm_mean` / `_norm_std` for zscore
- Users cannot switch normalization methods without retraining

## Proposed Solution

Allow passing `normalization` as an optional parameter to `score()` and `detect()` methods to override the model's default at inference time.

### Implementation Steps

1. **Update `AnomalyModel.score()` and `detect()` signatures:**
   ```python
   async def score(
       self,
       data: list[list[float]] | np.ndarray,
       threshold: float | None = None,
       normalization: str | None = None,  # NEW: override model's default
   ) -> list[AnomalyScore]:
   ```

2. **Update `_normalize_scores()` to accept override:**
   ```python
   def _normalize_scores(
       self, scores: np.ndarray, normalization_override: str | None = None
   ) -> np.ndarray:
       norm_method = normalization_override or self.normalization
       # ... rest of method uses norm_method instead of self.normalization
   ```

3. **Update server endpoints to pass normalization through:**
   - The `request.normalization` parameter should be passed to `model.score()` and `model.detect()`

4. **Handle cache key complexity:**
   - The current cache key includes normalization: `anomaly:{backend}:{normalization}:{model_id}`
   - Need to either:
     - (a) Remove normalization from cache key for inference operations, OR
     - (b) Add a helper function to find cached models regardless of normalization

### End State

```python
# Train model with standardization (default)
model = AnomalyModel(..., normalization="standardization")
await model.fit(data)

# Score with different normalizations - no retraining needed!
results_std = await model.score(test_data)  # Uses model's default
results_zscore = await model.score(test_data, normalization="zscore")  # Override
results_raw = await model.score(test_data, normalization="raw")  # Override
```

## Why This Matters

- Users can explore how different normalization methods affect anomaly detection
- Allows A/B testing of thresholds without retraining
- Model already stores all necessary statistics - just needs to expose the flexibility

## Complexity

Medium - The core changes to `AnomalyModel` are simple, but the server-side caching logic needs careful handling to ensure a model trained with one normalization can serve requests for other normalizations.

## Labels

enhancement, anomaly-detection
